app/scripts/controllers/metametrics.js

```diff
@@ -443,13 +443,26 @@ export default class MetaMetricsController {
     }
   }
 
+  // Method to track metrics event
+  // If an actionId is provided createEventFragment, finalizeEventFragment are used to track event,
+  // this is to achieve idempotent behaviour.
+  // If actionid is not provided this._trackEvent is used to track the event.
+  trackEvent(payload, options) {
+    if (options?.actionId) {
+      const fragment = this.createEventFragment({ ...payload, ...options });
```

Comment:
I forgot one tidbit here, we can't use the payload of trackEvent directly, we'll need to manipulate some things.

see shared/constants/metametrics for details but I think the only thing we need to change is `payload.event` -> `payload.successEvent`